sys/kern/vm_map.c

```diff
@@ -161,6 +161,46 @@ void vm_segment_destroy(vm_map_t *map, vm_segment_t *seg) {
   vm_segment_free(seg);
 }
 
+/* there are few cases:
+ * 1) we want to free the whole segment
+ * 2) we want to free the beginning of the segment and leave the end
+ * without changes
+ * 3) we want to free the end of the segment and leave the beginning
+ * without changes
+ * 4) we want to free the middle of the segment, so we want to split it to
+ * the new two segments with the hole between them */
```

Comment:
Imo this comments should be left by the code which they are describing.

Reply:
I changed code a little and this comment is no longer relevant, so I deleted it.